pkg/reconciler/v1alpha1/service/service.go

```diff
@@ -207,7 +207,21 @@ func (c *Reconciler) reconcile(ctx context.Context, service *v1alpha1.Service) e
 	}
 
 	// Update our Status based on the state of our underlying Route.
-	service.Status.PropagateRouteStatus(route.Status)
+	ss := &service.Status
+	ss.PropagateRouteStatus(&route.Status)
+
+	// Apply additional logic, once the generic data has been propagated.
+	switch {
```

Comment:
This is going to become quite large once you add release mode. MIght be worth pulling out into a function that takes route, config, and ss.